Formula/elasticsearch@7.6.rb

```diff
@@ -0,0 +1,150 @@
+class ElasticsearchAT76 < Formula
+  desc "Distributed search & analytics engine"
+  homepage "https://www.elastic.co/products/elasticsearch"
+  url "https://github.com/elastic/elasticsearch/archive/v7.6.2.tar.gz"
+  sha256 "6ff4871dcae6954e13680aefc196da574a59a36418d06a7e095550ce81a370f8"
+
+  head do
+    url "https://github.com/elasticsearch/elasticsearch.git"
+  end
+
+  depends_on "openjdk"
+
+  def cluster_name
+    "elasticsearch_#{ENV["USER"]}"
+  end
+
+  def install
+    # Build the package from source
+    system "./gradlew", ":distribution:archives:oss-no-jdk-darwin-tar:assemble"
+
+    mkdir "tar" do
+      # Extract the package to the tar directory
+      system "tar", "--strip-components=1", "-xf",
+        Dir["../distribution/archives/oss-no-jdk-darwin-tar/build/distributions/elasticsearch-oss-*.tar.gz"].first
+
+      # Install into package directory
+      libexec.install "bin", "config", "lib", "modules"
+    end
+
+    inreplace libexec/"bin/elasticsearch-env",
+              "if [ -z \"$ES_PATH_CONF\" ]; then ES_PATH_CONF=\"$ES_HOME\"/config; fi",
+              "if [ -z \"$ES_PATH_CONF\" ]; then ES_PATH_CONF=\"#{etc}/elasticsearch\"; fi"
+
+    # Set up Elasticsearch for local development:
+    inreplace "#{libexec}/config/elasticsearch.yml" do |s|
+      # 1. Give the cluster a unique name
+      s.gsub!(/#\s*cluster\.name\: .*/, "cluster.name: #{cluster_name}")
+
+      # 2. Configure paths
+      s.sub!(%r{#\s*path\.data: /path/to.+$}, "path.data: #{var}/lib/elasticsearch/")
+      s.sub!(%r{#\s*path\.logs: /path/to.+$}, "path.logs: #{var}/log/elasticsearch/")
+    end
+
+    inreplace "#{libexec}/config/jvm.options", %r{logs/gc.log}, "#{var}/log/elasticsearch/gc.log"
+
+    # Move config files into etc
+    (etc/"elasticsearch").install Dir[libexec/"config/*"]
+    (libexec/"config").rmtree
+
+    bin.install libexec/"bin/elasticsearch",
+                libexec/"bin/elasticsearch-keystore",
+                libexec/"bin/elasticsearch-plugin",
+                libexec/"bin/elasticsearch-shard"
+    bin.env_script_all_files(libexec/"bin", :JAVA_HOME => Formula["openjdk"].opt_prefix)
+  end
+
+  def post_install
+    # Make sure runtime directories exist
+    (var/"lib/elasticsearch").mkpath
+    (var/"log/elasticsearch").mkpath
+    ln_s etc/"elasticsearch", libexec/"config" unless (libexec/"config").exist?
+    (var/"elasticsearch/plugins").mkpath
+    ln_s var/"elasticsearch/plugins", libexec/"plugins" unless (libexec/"plugins").exist?
+    # fix test not being able to create keystore because of sandbox permissions
+    system bin/"elasticsearch-keystore", "create" unless (etc/"elasticsearch/elasticsearch.keystore").exist?
```

Comment:
Is this also useful for users? Or just for the test?

Reply:
A far as I understand this, we need a created keystore to run the tests. But because `etc/elasticsearch` is protected by the sandbox during test time, we need to create said keystore in `post_install`.

However I'm not sure how good and bad already creating a default keystore is for actual users of this formula.